### PR TITLE
feat(aws): More Redshift resources

### LINKED ISSUE #7919: AWS RedShift Additional Resources
Redshift Endpoint Access & Authorization
https://docs.aws.amazon.com/redshift/latest/APIReference/API_EndpointAccess.html
https://docs.aws.amazon.com/redshift/latest/APIReference/API_EndpointAuthorization.html
Redshift Datashare
https://docs.aws.amazon.com/redshift/latest/APIReference/API_DataShare.html
RedShift Events
https://docs.aws.amazon.com/redshift/latest/APIReference/API_Event.html

### PR DESCRIPTION
**Pull request by disq**
Resolves #7919

First two commits are clean ups. No PKs were added because I was unsure. No ARNs were made up.

`aws_redshift_events` can benefit from #9252

### CODE REVIEW

**Comment by hermanschaaf** [comment]
Maybe let's add `aws_redshift_events` to the list of configuration-like tables in https://github.com/cloudquery/cloudquery/blob/main/website/pages/docs/plugins/sources/aws/configuration.md#skipping-tables-with-configuration-parameters? It's getting a bit fuzzy what we should add there though to be honest.

File: plugins/source/aws/resources/services/redshift/events.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func Events() *schema.Table {
+	tableName := "aws_redshift_events"
+	return &schema.Table{
+		Name: tableName,
+		Description: `https://docs.aws.amazon.com/redshift/latest/APIReference/API_Event.html.
+
+Only events occurred in the last hour are returned.`,
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/redshift/events.go:20]: Hmm, this is quite a big caveat 😄 Is this an API limitation?

Comment (disq) [reply plugins/source/aws/resources/services/redshift/events.go:20]: Until we do stateful fetches it wouldn't be nice to pull all events (for the past 14 days, which is the api limit) again and again all the time. The api needs some time to go back (and an end time) with some source filtering options as well: https://docs.aws.amazon.com/redshift/latest/APIReference/API_DescribeEvents.html
one hour is the api default.

**Review by hermanschaaf — APPROVED** [review]
Looks good to me 👍 We may want to think about the PK for the events, but I don't know if the ID will be globally unique.

File: website/tables/aws/aws_redshift_events.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+| ------------- | ------------- |
+|_cq_source_name|String|
+|_cq_sync_time|Timestamp|
+|_cq_id (PK)|UUID|
+|_cq_parent_id|UUID|
+|account_id|String|
+|region|String|
+|date|Timestamp|
+|event_categories|StringArray|
+|event_id|String|
```
[the remark this answers is not in the document]
Comment (disq) [reply website/tables/aws/aws_redshift_events.md:23]: > Would event ID be able to serve as a PK?

Not sure how unique it is, or if you can access other accounts events etc.

[the remark this answers is not in the document]
Comment (hermanschaaf) [reply website/tables/aws/aws_redshift_events.md:23]: Yeah me neither 🤷 I think let's leave it as cq_id

**Pull request merged**